test/unit/hooks/highPingKick/index.test.js

```diff
@@ -0,0 +1,96 @@
+    sandbox.stub(sails.helpers.sdtdApi, 'getOnlinePlayers').returns(Promise.resolve(
+      [
+        {'steamid':'123456','entityid':97356,'ip':'1.1.1.1','name':'Fake Name','online':true,'position':{'x':81,'y':67,'z':-1030},'experience':-1,'level':3.63118052482605,'health':72,'stamina':79.4040069580078,'zombiekills':7,'playerkills':0,'playerdeaths':1,'score':7,'totalplaytime':0,'lastonline':'0001-01-01T00:00:00','ping':524}
+      ]
+    ));
+    sails.testServerConfig.pingKickEnabled = 1;
+    sandbox.stub(SdtdConfig, 'findOne').returns(Promise.resolve(sails.testServerConfig));
+
+    const highPingCount = new HighPingCount({ });
+
+    await highPingCount.handlePingCheck({
+      server: sails.testServer
+    });
+  });
+  it('Ignores user in whitelist', async () => {
+    sandbox.stub(sails.helpers.sdtdApi, 'getOnlinePlayers').returns(Promise.resolve(
+      [
+        {'steamid':sails.testPlayer.steamId,'entityid':97356,'ip':'1.1.1.1','name':'Fake Name','online':true,'position':{'x':81,'y':67,'z':-1030},'experience':-1,'level':3.63118052482605,'health':72,'stamina':79.4040069580078,'zombiekills':7,'playerkills':0,'playerdeaths':1,'score':7,'totalplaytime':0,'lastonline':'0001-01-01T00:00:00','ping':524}
+      ]
+    ));
+
+    sails.testServerConfig.pingKickEnabled = 1;
+    sails.testServerConfig.pingWhitelist = JSON.stringify([ sails.testPlayer.steamId ]);
+
+    sandbox.stub(SdtdConfig, 'findOne').returns(Promise.resolve(sails.testServerConfig));
+
+    const highPingCount = new HighPingCount({ });
+
+    await highPingCount.handlePingCheck({
+      server: sails.testServer
+    });
+  });
+  it('kicks a player who has too high of ping for too long', async () => {
+    sandbox.stub(sails.helpers.sdtdApi, 'getOnlinePlayers').returns(Promise.resolve(
+      [
+        {'steamid':sails.testPlayer.steamId,'entityid':97356,'ip':'1.1.1.1','name':'Fake Name','online':true,'position':{'x':81,'y':67,'z':-1030},'experience':-1,'level':3.63118052482605,'health':72,'stamina':79.4040069580078,'zombiekills':7,'playerkills':0,'playerdeaths':1,'score':7,'totalplaytime':0,'lastonline':'0001-01-01T00:00:00','ping':524}
+      ]
+    ));
+    const execSpy = sandbox.stub(sails.helpers.sdtdApi, 'executeConsoleCommand').returns(Promise.resolve());
+
+    sails.testServerConfig.pingKickEnabled = 1;
+    sails.testServerConfig.pingChecksToFail = 3;
+    sails.testServerConfig.pingWhitelist = JSON.stringify([ ]);
+
+    sandbox.stub(SdtdConfig, 'findOne').returns(Promise.resolve(sails.testServerConfig));
+
+    const highPingCount = new HighPingCount({ });
+
+    // first time
+    await highPingCount.handlePingCheck({
+      server: sails.testServer
+    });
+    // second time
+    await highPingCount.handlePingCheck({
+      server: sails.testServer
+    });
+    // third time
+    await highPingCount.handlePingCheck({
+      server: sails.testServer
+    });
+    // 4th time it should get kicked
```

Comment:
Perhaps add `expect(execSpy).to.have.callCount(0);` before the fourth time?

Lets say this fires on the 3rd time, but doesn't on the 4th that would be incorrect behaviour but this test wouldn't catch it

Reply:
True true. I'll add that